src/ts/multinonce.ts

```diff
@@ -1,69 +1,62 @@
-import { defaultAbiCoder, BigNumber, keccak256 } from "ethers/utils";
+import { defaultAbiCoder, BigNumber } from "ethers/utils";
 import { Wallet } from "ethers";
 import { ReplayProtectionAuthority } from "./replayprotectionauthority";
 import { Lock } from "@pisa-research/utils";
 
 export class MultiNonce extends ReplayProtectionAuthority {
-  indexTracker: Map<string, BigNumber>;
-  nonceTracker: Map<string, BigNumber>;
+  private index: BigNumber;
+  // Weird JS Bug. If BigNumber is the Key, then the lookup
+  // will fail. String iS OK.
+  private nonceTracker: Map<string, BigNumber>;
   lock: Lock;
```

Comment:
This lock should be readonly, you dont want people replacing it. also you can just set it here instead of the constructor. At the moment it's also public (the default), which makes the combination insecure as someone can override your lock externally.
```
private readonly lock = new Lock()
```

Reply:
made readonly